app/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataRecord.java

```diff
@@ -1,185 +1,154 @@
 package org.sagebionetworks.bridge.dynamodb;
 
-import org.sagebionetworks.bridge.json.DateTimeJsonDeserializer;
-import org.sagebionetworks.bridge.json.DateTimeJsonSerializer;
-import org.sagebionetworks.bridge.json.JsonUtils;
+import org.sagebionetworks.bridge.json.DateTimeToStringSerializer;
+import org.sagebionetworks.bridge.json.JodaDateTimeDeserializer;
+import org.sagebionetworks.bridge.json.LocalDateToStringSerializer;
 import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecordBuilder;
 
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
-import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.datatype.joda.deser.LocalDateDeserializer;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
 
-@DynamoDBTable(tableName = "HealthDataRecord2")
+/** DynamoDB implementation of {@link org.sagebionetworks.bridge.models.healthdata.HealthDataRecord}. */
+@DynamoDBTable(tableName = "HealthDataRecord3")
 public class DynamoHealthDataRecord implements HealthDataRecord, DynamoTable {
-
-    private static final String GUID_FIELD = "guid";
-    private static final String START_DATE_FIELD = "startDate";
-    private static final String END_DATE_FIELD = "endDate";
-    private static final String VERSION_FIELD = "version";
-    private static final String DATA_FIELD = "data";
-    
-    private String key;
-    private String guid;
-    private long startDate;
-    private long endDate;
     private JsonNode data;
+    private String healthCode;
```

Comment:
Yes, we have a secondary index on health code, and we could filter by schema. (Whether we want a specialized secondary index on both health code and schema can be easily answered in the future.)